django/template/defaulttags.py

```diff
@@ -10,6 +10,7 @@
 from django.conf import settings
 from django.utils import timezone
 from django.utils.html import conditional_escape, escape, format_html
+from django.utils.itercompat import is_iterable
```

Comment:
I'm curious as to why we still use this over `isinstance(…, collections.abc.Iterable)`? 🤔

Looking at Python's [documentation](https://docs.python.org/3.10/library/collections.abc.html) -- see `3)` in the first section -- it seems as though it should work perfectly fine. Is there something I'm missing?

It looks as though this was introduced in ad077ccbc03f11c3db3951aab110ec98c5907729 for compatibility with Jython which we no longer support… Perhaps it's time to deprecate `is_iterable()`?

Reply:
`is_iterable` would presumably be `True` for objects using the old `__getitem__` method, which I don't think is equivalent for `isinstance(..., Iterable)`:

> Checking `isinstance(obj, Iterable)` detects classes that are registered as `Iterable` or that have an `__iter__()` method, but it does not detect classes that iterate with the `__getitem__()` method. The only reliable way to determine whether an object is iterable is to call `iter(obj)`.

That may or may not matter these days (or my reading my be wrong 🤷)

Reply:
> It looks as though this was introduced in [ad077cc](https://github.com/django/django/commit/ad077ccbc03f11c3db3951aab110ec98c5907729) for compatibility with Jython which we no longer support… Perhaps it's time to deprecate `is_iterable()`?

Maybe it's time, but we should discuss it and do it separately. I've changed to `is_iterable()` to keep the code consistent for now.

Reply:
I opened ticket-34983 for this.